snarkos/store/rocksdb/tests.rs

```diff
@@ -14,185 +14,170 @@
+fn test_a_value_cannot_be_removed_twice() {
+    let test_map = setup_test_map(8);
+    test_map.insert(123456789, "123456789".to_string()).expect("Failed to insert");
+    test_map.remove(&123456789).expect("Failed to remove");
+
+    let expected_result = test_map.remove(&123456789);
+
+    assert!(expected_result.is_err());
 }
 
 #[test]
-fn test_export_import() {
-    let file = temp_file();
+#[serial]
+fn test_can_iter_on_pairs_after_inserting() {
+    let test_map = setup_test_map(9);
 
-    {
-        let storage = RocksDB::open(temp_dir(), 0).expect("Failed to open storage");
-        let mut map = storage.open_map::<u32, String>(DataID::Test).expect("Failed to open data map");
-
-        for i in 0..100 {
-            map.insert(i, i.to_string()).expect("Failed to insert");
-        }
+    test_map.insert(123456789, "123456789".to_string()).expect("Failed to insert");
+    let expected_result = test_map.iter().next().map(|(k, v)| (*k, v.to_string()));
 
-        storage.export(&file).expect("Failed to export storage");
-    }
+    assert_eq!(Some((123456789, "123456789".to_string())), expected_result);
+}
 
-    let storage = RocksDB::open(temp_dir(), 0).expect("Failed to open storage");
-    storage.import(&file).expect("Failed to import storage");
+#[test]
+#[serial]
+fn test_can_iter_on_keys_after_inserting() {
+    let test_map = setup_test_map(11);
 
-    let map = storage.open_map::<u32, String>(DataID::Test).expect("Failed to open data map");
+    test_map.insert(123456789, "123456789".to_string()).expect("Failed to insert");
+    let mut keys = test_map.keys();
+    let expected_result = keys.next().map(|k| *k);
+    let expected_none = keys.next();
 
-    for i in 0..100 {
-        assert_eq!(map.get(&i).expect("Failed to get").map(|v| v.to_string()), Some(i.to_string()));
-    }
+    assert_eq!(Some(123456789), expected_result);
+    assert!(expected_none.is_none());
 }
 
 #[test]
-fn test_scalar_mul() {
-    type CurrentNetwork = Testnet3;
-
-    let rng = &mut test_crypto_rng();
+#[serial]
+fn test_can_iter_on_values_after_inserting() {
+    let test_map = setup_test_map(12);
 
-    const ITERATIONS: u32 = 1_000_000u32;
+    test_map.insert(123456789, "123456789".to_string()).expect("Failed to insert");
+    let mut values = test_map.values();
+    let expected_result = values.next().map(|v| v.to_string());
+    let expected_none = values.next();
 
-    let storage = RocksDB::open(temp_dir(), 0).expect("Failed to open storage");
-    let mut map = storage.open_map(DataID::Test).expect("Failed to open data map");
+    assert_eq!(Some("123456789".to_string()), expected_result);
+    assert!(expected_none.is_none());
+}
 
-    // Sample `ITERATION` random field elements to store.
-    for i in 0..ITERATIONS {
-        let value = Scalar::<CurrentNetwork>::rand(rng);
-        map.insert(i, value).expect("Failed to insert");
+#[test]
+#[serial]
+fn test_reopen() {
+    {
+        let test_map = setup_test_map(13);
```

Comment:
we should probably be picking the ID at random in all the tests if we want to try as many as possible out in different conditions

Reply:
Do you mean the `network_id` or the `DataID`? If you mean the frist one I think its ok, and if you mean the `DataID` what do you think about iterate through all the enum values? (if this ends to be too slow, we'll go with the random)

Reply:
The data id; the network id should probably always be the current one (though right now it would interfere with the non-test db). As for iterating, I'd suggest having a `static` ID that starts with a random value and is bumped with each test and wraps around, but it likely wouldn't work due to the use of `serial`; just doing all IDs in every test is redundant, it's good enough that a few get tested with every run.